### PR TITLE
fixes #127 Provided service type LoadBalancer

### LINKED ISSUE #127: Support for Ingress type 'LodBalancer'
Currently the ingress template only supports CluterIP and NodePort. Could you please add support for LoadBalancer

### PR DESCRIPTION
**Pull request by billett83**
Added support for service type LoadBalancer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,16 @@ spec:
...
 ```
 
+  * LoadBalancer
+
+```yaml
+---
+spec:
+  ...
+  tower_ingress_type: Loadbalancer
+  tower_ingress_protocal: http
```
Comment (shanemcd) [inline README.md]:
```suggestion
  tower_ingress_protocol: http
```

File: README.md [hunk 2]
```diff
@@ -129,6 +129,16 @@ spec:
   tower_hostname: awx.mycompany.com
 ```
 
+  * LoadBalancer
+
+```yaml
+---
+spec:
+  ...
+  tower_ingress_type: Loadbalancer
```
Comment (shanemcd) [inline README.md]:
```suggestion
  tower_ingress_type: LoadBalancer
```

File: ansible/instantiate-awx-deployment.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@
           spec:
             tower_admin_user: test
             tower_admin_email: test@example.com
-            tower_ingress_type: "{{ tower_ingress_type | default(omit) }}"  # Either Route or Ingress
+            tower_ingress_type: "{{ tower_ingress_type | default(omit) }}"  # Either Route, Ingress or Loadbalancer
```
Comment (shanemcd) [inline ansible/instantiate-awx-deployment.yml]:
```suggestion
            tower_ingress_type: "{{ tower_ingress_type | default(omit) }}"  # Either Route, Ingress or LoadBalancer
```

File: ansible/templates/crd.yml.j2 [hunk 4]
```diff
@@ -62,12 +62,28 @@ spec:
                     - ingress
                     - Route
                     - route
+                    - Loadbalancer
```
Comment (shanemcd) [inline ansible/templates/crd.yml.j2]:
```suggestion
                    - LoadBalancer
```

File: deploy/awx-operator.yaml [hunk 5]
```diff
@@ -207,12 +207,28 @@ spec:
                     - ingress
                     - Route
                     - route
+                    - Loadbalancer
```
Comment (shanemcd) [inline deploy/awx-operator.yaml]:
```suggestion
                    - LoadBalancer
```

File: deploy/crds/awx_v1beta1_crd.yaml [hunk 6]
```diff
@@ -62,12 +62,28 @@ spec:
                     - ingress
                     - Route
                     - route
+                    - Loadbalancer
```
Comment (shanemcd) [inline deploy/crds/awx_v1beta1_crd.yaml]:
```suggestion
                    - LoadBalancer
```

File: deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml [hunk 7]
```diff
@@ -127,6 +128,23 @@ spec:
         - urn:alm:descriptor:com.tectonic.ui:advanced
         - urn:alm:descriptor:io.kubernetes:Secret
         - urn:alm:descriptor:com.tectonic.ui:fieldDependency:tower_ingress_type:Ingress
+      - displayName: Tower LoadBalancer Annotations
+        path: tower_loadbalancer_annotations
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:advanced
+        - urn:alm:descriptor:com.tectonic.ui:text
+        - urn:alm:descriptor:com.tectonic.ui:fieldDependency:tower_ingress_type:Loadbalancer
```
Comment (shanemcd) [inline deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml]:
```suggestion
        - urn:alm:descriptor:com.tectonic.ui:fieldDependency:tower_ingress_type:LoadBalancer
```

**Review by shanemcd — COMMENTED** [review]
Did some nitpicking but overall this is awesome. Thanks for this.

File: roles/installer/templates/tower_service.yaml.j2 [hunk 8]
```diff
@@ -6,20 +6,34 @@ metadata:
...
+{% if tower_ingress_type | lower == 'loadbalancer' and tower_loadbalancer_protocol | lower == 'https' %}
+    - port: 443
+      protocol: TCP
+      targetPort: 8052
+      name: https
 {% endif %}
   selector:
     app: '{{ deployment_type }}'
-{% if tower_ingress_type != "none" %}
+{% if tower_ingress_type == "loadbalancer" %}
```
Comment (Spredzy) [inline roles/installer/templates/tower_service.yaml.j2]: Missing a `| lower` here

File: deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml [hunk 9]
```diff
@@ -115,6 +115,7 @@ spec:
         - urn:alm:descriptor:com.tectonic.ui:select:none
         - urn:alm:descriptor:com.tectonic.ui:select:Ingress
         - urn:alm:descriptor:com.tectonic.ui:select:Route
+        - urn:alm:descriptor:com.tectonic.ui:select:Loadbalancer
```
Comment (Spredzy) [inline deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml]: s/Loadbalancer/LoadBalancer/

File: deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml [hunk 10]
```diff
@@ -127,6 +128,23 @@ spec:
...
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:advanced
+        - urn:alm:descriptor:com.tectonic.ui:text
+        - urn:alm:descriptor:com.tectonic.ui:fieldDependency:tower_ingress_type:LoadBalancer
+      - displayName: Tower LoadBalancer Protocol
+        path: tower_loadbalancer_protocol
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:advanced
+        - urn:alm:descriptor:com.tectonic.ui:select:http
+        - urn:alm:descriptor:com.tectonic.ui:select:https
```
Comment (Spredzy) [inline deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml:142]: Need to add `urn:alm:descriptor:com.tectonic.ui:fieldDependency:tower_ingress_type:LoadBalancer`

File: deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml [hunk 11]
```diff
@@ -127,6 +128,23 @@ spec:
...
+      - displayName: Tower LoadBalancer Port
+        path: tower_loadbalancer_port
+        x-descriptors:
+        - urn:alm:descriptor:com.tectonic.ui:advanced
+        - urn:alm:descriptor:com.tectonic.ui:number
```
Comment (Spredzy) [inline deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml:148]: Need to add `urn:alm:descriptor:com.tectonic.ui:fieldDependency:tower_ingress_type:LoadBalancer`

File: deploy/olm-catalog/awx-operator/manifests/awx.ansible.com_awxs_crd.yaml [hunk 12]
```diff
@@ -82,7 +82,23 @@ spec:
                 - ingress
                 - Route
                 - route
+                - Loadbalancer
```
Comment (Spredzy) [inline deploy/olm-catalog/awx-operator/manifests/awx.ansible.com_awxs_crd.yaml]: s/Loadbalancer/LoadBalancer/

**Review by Spredzy — CHANGES_REQUESTED** [review]
Minor comments, overall looks good! Thank you.

**Comment by billett83** [comment]
Thank you both. I have reviewed and committed your suggested changed @Spredzy

File: deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml [hunk 13]
```diff
@@ -115,7 +115,7 @@ spec:
         - urn:alm:descriptor:com.tectonic.ui:select:none
         - urn:alm:descriptor:com.tectonic.ui:select:Ingress
         - urn:alm:descriptor:com.tectonic.ui:select:Route
-        - urn:alm:descriptor:com.tectonic.ui:select:Loadbalancer
+        - urn:alm:descriptor:com.tectonic.ui:select:LoadB alancer
```
Comment (Spredzy) [inline deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml]: nit: extra space here

Comment (billett83) [reply deploy/olm-catalog/awx-operator/manifests/awx-operator.clusterserviceversion.yaml]: Apologies

**Review by Spredzy — APPROVED** [review]
Thank you !

**Pull request merged**